src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -332,15 +332,12 @@ public async Task<SyncTriggersPayload> GetSyncTriggersPayload()
             var functionDetails = await WebFunctionsManager.GetFunctionMetadataResponse(listableFunctions, hostOptions, _hostNameProvider);
             result.Add("functions", new JArray(functionDetails.Select(p => JObject.FromObject(p))));
 
-            // TEMP: refactor this code to properly add extensions in all scenario(#7394)
-            // Add the host.json extensions to the payload
-            if (_environment.IsKubernetesManagedHosting())
+            JObject extensionsPayload = null;
+            extensionsPayload = _environment.IsKubernetesManagedHosting() ? await GetHostJsonExtensionsForKubernetesAsync(_applicationHostOptions, _logger) : await GetHostJsonExtensionsAsync(_applicationHostOptions, _logger);
```

Comment:
nit: multi-line is more readable
```suggestion
            extensionsPayload = _environment.IsKubernetesManagedHosting() ? 
                await GetHostJsonExtensionsForKubernetesAsync(_applicationHostOptions, _logger) : 
                await GetHostJsonExtensionsAsync(_applicationHostOptions, _logger);
```